rust/src/coordinator.rs

```diff
@@ -574,85 +612,82 @@ mod tests {
         let dict_sum = auxiliary_sum(coord.min_sum);
         for (encr_pk, ephm_pk) in dict_sum.iter() {
             assert!(!coord.has_enough_sum_participants());
-            coord.update_dict_sum(encr_pk, ephm_pk);
+            coord.add_sum_participant(encr_pk, ephm_pk);
         }
         assert_eq!(coord.dict_sum, dict_sum);
-        assert!(coord.dict_seed.is_empty());
+        assert!(coord.dict_update.is_empty());
         assert!(coord.has_enough_sum_participants());
 
         coord.maybe_transition(); // finish the sum phase
         assert_eq!(coord.phase, Phase::Update);
         assert_eq!(
-            coord.dict_seed,
+            coord.dict_update,
             dict_sum
                 .iter()
                 .map(|(encr_pk, _)| (*encr_pk, HashMap::new()))
                 .collect(),
         );
     }
 
+    fn generate_update(dict_sum: &SumDict) -> (UpdateParticipantPublicKey, SeedDict) {
+        let seed = randombytes(32);
+        let pk = box_::PublicKey::from_slice(&randombytes(32)).unwrap();
+        let seed_dict = dict_sum
+            .iter()
+            .map(|(sum_encr_pk, sum_ephm_pk)| (*sum_encr_pk, sealedbox::seal(&seed, sum_ephm_pk)))
+            .collect::<SeedDict>();
+        (pk, seed_dict)
+    }
+
     fn auxiliary_update(
         min_sum: usize,
         min_update: usize,
     ) -> (
-        HashMap<box_::PublicKey, box_::PublicKey>,
-        Vec<(box_::PublicKey, HashMap<box_::PublicKey, Vec<u8>>)>,
-        HashMap<box_::PublicKey, HashMap<box_::PublicKey, Vec<u8>>>,
+        SumDict,
+        Vec<(UpdateParticipantPublicKey, SeedDict)>,
+        UpdateDict,
     ) {
         let dict_sum = auxiliary_sum(min_sum);
 
-        let updates = iter::repeat_with(|| {
-            let seed = randombytes(32);
-            let upd_encr_pk = box_::PublicKey::from_slice(&randombytes(32)).unwrap();
-            let upd_dict_seed = dict_sum
+        let updates = iter::repeat_with(|| generate_update(&dict_sum))
+            .take(min_update)
+            .collect::<Vec<(UpdateParticipantPublicKey, SeedDict)>>();
+
+        let mut dict_update = UpdateDict::new();
+        for sum_encr_pk in dict_sum.keys() {
```

Comment:
@janpetschexain sorry, I know you like combinators, but that was just too much for my brain so I converted the outer loop into an actual `for` loop :smile:

Reply:
no worries, i probably overdid it in some places^^